libnm-core/nm-setting-tc-config.c

```diff
@@ -1087,15 +1087,17 @@ set_property (GObject *object, guint prop_id,
 	switch (prop_id) {
 	case PROP_QDISCS:
 		g_ptr_array_unref (self->qdiscs);
-		self->qdiscs = _nm_utils_copy_array (g_value_get_boxed (value),
-		                                     (NMUtilsCopyFunc) nm_tc_qdisc_dup,
-		                                     (GDestroyNotify) nm_tc_qdisc_unref);
+		self->qdiscs = _nm_utils_copy_array_dedup (g_value_get_boxed (value),
```

Comment:
the problem is, when NetworkManager receives a setting on D-Bus, it will call this property getter, which silently drops duplicates. Wouln't it be better, if setting such properties is allowed in the first moment (because, you don't know what you will receive when parsing the D-Bus variant). Only in the next step, `nm_connection_verify()` either rejects duplicates with error, or `nm_connection_normalize()` normalizes them. So, one of `NM_SETTING_VERIFY_NORMALIZABLE`, `NM_SETTING_VERIFY_NORMALIZABLE_ERROR`, or `NM_SETTING_VERIFY_ERROR`

At this point (where OVS support is almost new API), I would just flag them as error. Not trying to normalize anything. The user did something wrong, don't try to fix it silently.